src/bin/pg_autoctl/pgsql.c

```diff
@@ -2204,7 +2205,7 @@ pgsql_get_postgres_metadata(PGSQL *pgsql,
 		" coalesce(rep.sync_state, '') as sync_state,"
 		" case when pg_is_in_recovery()"
 		" then coalesce(pg_last_wal_receive_lsn(), pg_last_wal_replay_lsn())"
-		" else pg_current_wal_lsn()"
+		" else pg_current_wal_flush_lsn()"
```

Comment:
Could we document (maybe at least on the PR) for the reason of this change?

Reply:
That's because the `pg_current_wal_lsn` gets an LSN that is not known to have been flushed on-disk yet, whereas `pg_current_wal_flush_lsn` embeds that guarantee. I guess in an ideal world that change would have happened on its own PR, but that's how it is now.